handler/src/main/java/io/netty/handler/ssl/ReleasableOpenSslEngine.java

```diff
@@ -0,0 +1,1988 @@
+    }
+
+    /**
+     * Converts the specified OpenSSL cipher suite to the Java cipher suite.
+     */
+    private String toJavaCipherSuite(String openSslCipherSuite) {
+        if (openSslCipherSuite == null) {
+            return null;
+        }
+
+        String prefix = toJavaCipherSuitePrefix(SSL.getVersion(ssl));
+        return CipherSuiteConverter.toJava(openSslCipherSuite, prefix);
+    }
+
+    /**
+     * Converts the protocol version string returned by {@link SSL#getVersion(long)} to protocol family string.
+     */
+    private static String toJavaCipherSuitePrefix(String protocolVersion) {
+        final char c;
+        if (protocolVersion == null || protocolVersion.length() == 0) {
+            c = 0;
+        } else {
+            c = protocolVersion.charAt(0);
+        }
+
+        switch (c) {
+            case 'T':
+                return "TLS";
+            case 'S':
+                return "SSL";
+            default:
+                return "UNKNOWN";
+        }
+    }
+
+    @Override
+    public final void setUseClientMode(boolean clientMode) {
+        if (clientMode != this.clientMode) {
+            throw new UnsupportedOperationException();
+        }
+    }
+
+    @Override
+    public final boolean getUseClientMode() {
+        return clientMode;
+    }
+
+    @Override
+    public final void setNeedClientAuth(boolean b) {
+        setClientAuth(b ? ClientAuth.REQUIRE : ClientAuth.NONE);
+    }
+
+    @Override
+    public final boolean getNeedClientAuth() {
+        return clientAuth == ClientAuth.REQUIRE;
+    }
+
+    @Override
+    public final void setWantClientAuth(boolean b) {
+        setClientAuth(b ? ClientAuth.OPTIONAL : ClientAuth.NONE);
+    }
+
+    @Override
+    public final boolean getWantClientAuth() {
+        return clientAuth == ClientAuth.OPTIONAL;
+    }
+
+    private void setClientAuth(ClientAuth mode) {
+        if (clientMode) {
+            return;
+        }
+        synchronized (this) {
+            if (clientAuth == mode) {
+                // No need to issue any JNI calls if the mode is the same
+                return;
+            }
+            switch (mode) {
+                case NONE:
+                    SSL.setVerify(ssl, SSL.SSL_CVERIFY_NONE, OpenSslContext.VERIFY_DEPTH);
+                    break;
+                case REQUIRE:
+                    SSL.setVerify(ssl, SSL.SSL_CVERIFY_REQUIRE, OpenSslContext.VERIFY_DEPTH);
+                    break;
+                case OPTIONAL:
+                    SSL.setVerify(ssl, SSL.SSL_CVERIFY_OPTIONAL, OpenSslContext.VERIFY_DEPTH);
+                    break;
+            }
+            clientAuth = mode;
+        }
+    }
+
+    @Override
+    public final void setEnableSessionCreation(boolean b) {
+        if (b) {
+            throw new UnsupportedOperationException();
+        }
+    }
+
+    @Override
+    public final boolean getEnableSessionCreation() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) A getX() method which returns a boolean should be named isX() [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3ABooleanGetMethodName)